code/game/objects/effects/landmarks.dm

```diff
@@ -458,26 +458,27 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/start/new_player)
 /obj/effect/landmark/ruin
 	var/datum/map_template/ruin/ruin_template
 
-/obj/effect/landmark/ruin/New(loc, my_ruin_template)
+/obj/effect/landmark/ruin/Initialize(mapload, my_ruin_template)
 	name = "ruin_[GLOB.ruin_landmarks.len + 1]"
-	..(loc)
+	. = ..(mapload)
 	ruin_template = my_ruin_template
 	GLOB.ruin_landmarks |= src
+	return ..()
```

Comment:
This whole section results in dual parent calls, see whichever one is to be kept.